charts/jenkins/templates/_helpers.tpl

```diff
@@ -109,6 +109,7 @@ jenkins:
     enabled: true
   mode: {{ .Values.controller.executorMode }}
   numExecutors: {{ .Values.controller.numExecutors }}
+  labelString: "{{ join " " .Values.controller.customJenkinsLabels }}"
```

Comment:
this is documented as an object, I think docs should be updated to show it's an array?

also this should probably be omitted when empty and not an empty string.

An empty string will clear out anything that was previously there

Reply:
Good catch, updated the docs

As for the second portion, I actually just added in the "functionality" to clear it out today when that's empty because I noticed that we do that in other places inside of `_helpers.tpl`, for example, with portions of the default kubernetes agents: https://github.com/hak33m16/helm-charts/blob/bugfix/371-custom-jenkins-labels-not-working/charts/jenkins/templates/_helpers.tpl#L145 If somebody were to add custom configuration on top of this, and then disable it, they'd lose all of their changes. It also would feel a little wonky IMO to remove the entry and see that my labels persisted, and find out that I instead have to specify a value for the labels array of `[""]`

It feels a bit like a difficult line to walk, and I always run into this issue when building on top of configuration (differentiating between when a value should NOOP or overwrite), but in my mind at least, I expect users to exclusively manage their Jenkins instance through JCasC when it's enabled

If you disagree with that ideology, I completely understand, I don't fully understand how the Jenkins Helm chart is expected to used, but I wanted to at least throw out _why_ I changed that

Reply:
You should be able to differentiate between empty string and null.

@torstenwalter any thoughts?

Reply:
This is the best thing I've found: https://github.com/Masterminds/sprig/issues/53#issuecomment-483414063

Not really self explaining.

Reply:
I think I may not have explained myself very well, sorry about. I more meant that it can be confusing when leaving it null NOOPs rather than forcing it to be empty (e.g. the inconsistent behavior I was linking). If the consensus is that's the route we'd like to go, I can definitely do that, will push an update/fix these conflicts later tonight

Reply:
I would set the default to an empty string or leave it empty in `values.yaml`.
In addition I would surround the line above with an if statement.

Reply:
Sorry about the embarrassing amount of commits that took if you guys got notified, I mimicked the CI environment exactly on my local box (exact version of `helm`, `ct`, and `helm test` installed as in the git workflow) and I still couldn't get them to run. I'd just run into arbitrary errors like being unable to test local charts easily with `helm test` or in the case of `ct`:
```
$ ct list-changed --config ct.yaml
Error: Invalid Semantic Version
Invalid Semantic Version

$ ct lint --config ct.yaml
Linting charts...
Error: Error loading configuration: 'chart_schema.yaml' neither specified nor found in default locations
Error loading configuration: 'chart_schema.yaml' neither specified nor found in default locations
```

I also didn't see any docs on how I should set them up/run them, not sure if I missed anything, but it might be helpful for new contributors